src/lashup_kv.erl

```diff
@@ -229,72 +231,116 @@ init_db(Nodes) ->
   mnesia:change_table_copy_type (schema, node(), disc_copies), % If the node was already running
   {ok, _} = application:ensure_all_started(mnesia),
   ExistingTables = mnesia:system_info(tables),
-  Tables = [kv],
+  Tables = [kv2, nclock],
   TablesToCreate = Tables -- ExistingTables,
   lists:foreach(fun create_table/1, TablesToCreate),
-  ok = mnesia:wait_for_tables(Tables, infinity).
+  ok = mnesia:wait_for_tables(Tables, infinity),
+  maybe_upgrade_table(ExistingTables).
 
-create_table(kv) ->
-  {atomic, ok} =  mnesia:create_table(kv, [
-    {attributes, record_info(fields, kv)},
+create_table(Table) ->
+  {atomic, ok} =  mnesia:create_table(Table, [
+    {attributes, get_record_info(Table)},
     {disc_copies, [node()]},
     {type, set}
   ]).
 
+get_record_info(kv2) ->
+  record_info(fields, kv2);
+get_record_info(nclock) ->
+  record_info(fields, nclock).
 
--spec(mk_write_fun(Key :: key(), OldVClock :: riak_dt_vclock:vclock() | undefined, Op :: riak_dt_map:map_op())
-      -> (fun())).
-mk_write_fun(Key, OldVClock, Op) ->
+maybe_upgrade_table(ExistingTables) ->
```

Comment:
Shouldn't we crash (or exit) if upgrade_table fails? I see we'll return a value, aborted, but I don't see where we check that? Maybe I'm just missing something.